File: clients/ios-native/TrailNoteApp/Core/Network/ApiEnvelope.swift
import Foundation

struct ApiEnvelope<T: Decodable>: Decodable {
    let success: Bool
    let code: String
    let message: String
    let data: T
}

struct PageEnvelope<T: Decodable>: Decodable {
    let records: [T]
    let total: Int
    let page: Int
    let pageSize: Int
}

File: clients/android-native/app/src/main/java/com/trailnote/android/core/network/ApiEnvelope.kt
package com.trailnote.android.core.network

data class ApiEnvelope<T>(
    val success: Boolean,
    val code: String,
    val message: String,
    val data: T,
)

data class PageEnvelope<T>(
    val records: List<T>,
    val total: Int,
    val page: Int,
    val pageSize: Int,
)
